include/eve/module/core/regular/impl/simd/arm/neon/convert.hpp

```diff
@@ -11,171 +11,332 @@
+          return wide<U,fixed<4>>(vmovn_u32(eve::combine(v,zero(as(v))))).slice(lower_);
+        }
+        else
+        {
+          return convert( eve::combine(v,zero(as(v))), tgt ).slice(lower_);
+        }
+      }
     }
-    //==============================================================================================
-    // uint32 -> ?
-    else if constexpr( catin == category::uint32x2 )
+    else  if constexpr( c_i == category::int32x4  )
     {
-           if constexpr( catou == category::float32x2 )        return vcvt_f32_u32(v0);
-      else if constexpr( catou == category::uint64x2 )         return vmovl_u32(v0);
-      else                                                     return convert_(EVE_RETARGET(simd_), v0, tgt);
+            if constexpr( c_o == category::float32x4  ) return vcvtq_f32_s32(v);
+      else  if constexpr( c_o == category::int16x4    ) return vmovn_s32(v);
+      else  if constexpr( std::integral<U> && sizeof(U) == 8   )
+      {
+        using type64 = typename wide<T,N>::template rebind<std::int64_t>;
+        auto [l,h] = v.slice(); // 2x i32x2
+        return wide<U,N>(type64(vmovl_s32(l)),type64(vmovl_s32(h)));
+      }
     }
-    else if constexpr( catin == category::uint32x4 )
+    else  if constexpr( c_i == category::uint32x4 )
     {
-           if constexpr( catou == category::float32x4 )        return vcvtq_f32_u32(v0);
-      else if constexpr( catou == category::uint16x4 )         return vmovn_u32(v0);
-      else                                                     return convert_(EVE_RETARGET(simd_), v0, tgt);
+            if constexpr( c_o == category::float32x4 )  return vcvtq_f32_u32(v);
+      else  if constexpr( c_o == category::uint16x4   )  return vmovn_u32(v);
+      else  if constexpr( std::integral<U> && sizeof(U) == 8   )
+      {
+        using type64 = wide<upgrade_t<T>,typename N::split_type>;
+        auto [l,h] = v.slice(); // 2x u32x2
+        return wide<U,N>(type64(vmovl_u32(l)),type64(vmovl_u32(h)));
+      }
     }
-    //==============================================================================================
-    // int16 -> ?
-    else if constexpr( catin == category::int16x4 )
+  }
+
+  //================================================================================================
+  // convert: (u)int16 -> U
+  template<integral_scalar_value T, typename N, real_scalar_value U>
+  EVE_FORCEINLINE wide<U, N>  convert_impl( EVE_SUPPORTS(neon128_)
+                                          , wide<T, N> const& v, as<U> const &tgt
+                                          ) noexcept
+      requires arm_abi<abi_t<T, N>> && (sizeof(T) == 2)
+  {
+    constexpr auto c_o = categorize<wide<U, N>>();
+    constexpr auto c_i = categorize<wide<T, N>>();
+
+    // (un)signed integers is handled by signed<->unsigned auto-conversion
+    // Conversion to double is emulated
+    if constexpr( std::same_as<double,U> ) return map(convert,v,tgt);
```

Comment:
Why not go to 32 bit ints?

Reply:
Seems lik for floats you also double in size. Just double in size for anything float?

Reply:
Fixed